### ISSUE TITLE
🐛 BUG:

### ISSUE THREAD

**Issue by MarceloRangel74** [slackhq/nebula#902]
### What version of `nebula` are you using?

1.7.2

### What operating system are you using?

Windows

### Describe the Bug

**This is a security issue**, but I can't report it on HackerOne.

After downloading the Windows release (https://github.com/slackhq/nebula/releases/download/v1.7.2/nebula-windows-amd64.zip), **Windows Defender** warned that it **found a trojan** in the downloaded file: **Win32/Vigorf.A**.

I also tried with the 1.7.1 version, and this one appears to be normal.

**Comment by nbrownus** [comment]
We are aware and working with Microsoft as best we can to resolve the false positive. You can get around it by building v1.7.2 from source for the time being.

**Comment by IanVS** [comment]
From @aarzhanov:

>The problem is that different anti-virus programs detect the nebula's executable file as containing a Trojan/Malware/etc.
>
>This is a [link](https://www.virustotal.com/gui/file/e65b7de82a4d99b8c6657ffaf4c0437a4c576ab3e3ceca022fbdf45fae438b03/detection) with scan results for release v1.7.2 - nebula-windows-amd64.zip
>
>This is a [link](https://www.virustotal.com/gui/file/e4e349f23ff7137c5e749c8a3b32631956aff2d88cef09254b02bbdd100e7b9c) with scan results for release v1.7.2 - nebula-darwin.zip
>
>If you build the executable file yourself, this does not solve the problem. Antiviruses such as McAfee still identify the file as containing a Trojan.
>
>Starting with version 1.5.0, the file is identified as containing a Trojan on virustotal.com.
>
>It seems to me that the seriousness of this problem is underestimated. This leads to quite serious problems if we imagine a situation where the nebula is used on a large number of Windows machines and access to the RDP is provided through a mesh network. If the antivirus is triggered, we lose access to the machine if it was provided only through the mesh network.

**Comment by alexwoellhaf** [comment]
> From @aarzhanov:
> 
> > The problem is that different anti-virus programs detect the nebula's executable file as containing a Trojan/Malware/etc.
> > This is a [link](https://www.virustotal.com/gui/file/e65b7de82a4d99b8c6657ffaf4c0437a4c576ab3e3ceca022fbdf45fae438b03/detection) with scan results for release v1.7.2 - nebula-windows-amd64.zip
> > This is a [link](https://www.virustotal.com/gui/file/e4e349f23ff7137c5e749c8a3b32631956aff2d88cef09254b02bbdd100e7b9c) with scan results for release v1.7.2 - nebula-darwin.zip
> > If you build the executable file yourself, this does not solve the problem. Antiviruses such as McAfee still identify the file as containing a Trojan.
> > Starting with version 1.5.0, the file is identified as containing a Trojan on virustotal.com.
> > It seems to me that the seriousness of this problem is underestimated. This leads to quite serious problems if we imagine a situation where the nebula is used on a large number of Windows machines and access to the RDP is provided through a mesh network. If the antivirus is triggered, we lose access to the machine if it was provided only through the mesh network.

I haven't done a deep dive yet but it appears that it is not being flagged for behavioral reasons, so I would assume its just hitting on signatures.  I'll try to find some time to build it from source with some extra random functionality added to see what it takes to get around the detection.

**Comment by TyIsI** [comment]
As @IanVS closed issue #951 as duplicate, I figured I would point out that that issue occurred with version 1.7.1 and `nebula-cert.exe` as opposed to version 1.7.2, and `nebula.exe`. (It's an easy oversight.)

**Comment by JohnMaguire** [comment]
@TyIsI Thanks for the clarification. This bug is tracking the issue across many versions of Nebula and both binaries.

**Comment by rnalrd** [comment]
It appears that several AVs are affected, not only Windows Defender: https://www.virustotal.com/gui/file/e65b7de82a4d99b8c6657ffaf4c0437a4c576ab3e3ceca022fbdf45fae438b03

**Comment by aarzhanov** [comment]
> It appears that several AVs are affected, not only Windows Defender: https://www.virustotal.com/gui/file/e65b7de82a4d99b8c6657ffaf4c0437a4c576ab3e3ceca022fbdf45fae438b03

Exactly. And not only version 1.7.2. It seems to me that the name of the issue needs to be corrected...

**Comment by JohnMaguire** [comment]
We are continuing on work to solve this issue. Unfortunately, we have been unable to resolve this issue via contact with security vendors.

We are working to get an EV cert we can using for release signing, which we hope will mitigate the issue (#286.)

Please be patient while we continue to work to resolve this issue. We're aware of the challenges it poses in Windows environments.

**Comment by aarzhanov** [comment]
> We are continuing on work to solve this issue. Unfortunately, we have been unable to resolve this issue via contact with security vendors.
> 
> We are working to get an EV cert we can using for release signing, which we hope will mitigate the issue (#286.)
> 
> Please be patient while we continue to work to resolve this issue. We're aware of the challenges it poses in Windows environments.

Thank you for your efforts.  I don't think signing with a certificate will help with this problem.  We sign our MSI package and this does not help in the situation with antiviruses.  Of course, I will double-check what and how we sign when generating the MSI package and write the results here.

**Comment by JohnMaguire** [comment]
@aarzhanov Thanks for the comment. I went searching and [found this article](https://www.microsoft.com/en-us/security/blog/2018/08/16/partnering-with-the-industry-to-minimize-false-positives/)  which mentions that signing files detected as malware will hurt the certificate's reputation - I appreciate you mentioning this was unlikely to solve the problem before we went down that path.

> We use the reputation of digital certificates to help determine the reputation of files signed by them. The reverse is also true: we use the reputation of digitally signed files to determine the reputation of the digital certificates they are signed with. One of the most effective ways for developers to reduce the chances of their software being detected as malware is it to digitally sign files with a reputable certificate.

**Comment by JohnMaguire** [comment]
In happier news, we were able to get in contact with the Microsoft Defender team and they've removed the indicators that were causing false positives for Nebula v1.7.2.

After reviewing VirusTotal scans, v1.7.2 is the only version of Nebula detected by Microsoft Defender. Other AV products on may still be flagging Nebula v1.7.2 or earlier. That being said, we hope that Microsoft's removal of the Nebula software from their detection database will improve these false positives over time.

**According to Microsoft, the updated security definitions should be available for update in the next 24-48 hours.**

I will leave this ticket open until we have confirmed that the security updates have rolled out and Microsoft Defender is no longer detecting Nebula v1.7.2.

If you are affected by a false positive caused by another antivirus software, please open a new ticket describing the security product and version of Nebula affected and we'll do our best to follow up.

**Comment by JohnMaguire** [comment]
Just another update, per my own testing - it does seem that Defender is no longer detecting nebula.exe and nebula-cert.exe from inside the [release ZIP file](https://github.com/slackhq/nebula/releases/download/v1.7.2/nebula-windows-amd64.zip), but the ZIP file itself is still causing false positives, which makes download & install a challenge.

Additionally, I noted that when downloading any of the three files via MS Edge, it is still detecting false positives.

We are continuing to communicate with Microsoft to resolve the issue.

**Comment by nbrownus** [comment]
I am not longer having issues with 1.7.2 amd64 as of security intelligence version 1.397.548.0. Can anyone else try downloading the release and confirm?

**Comment by JohnMaguire** [comment]
Windows Defender should no longer be detecting Nebula v1.7.2 - ZIP files and EXEs. MS Edge is continuing to report them as malware, so please use a different browser to download Nebula until it is updated.

If you're still having issues, please ensure you have at least `Security Intelligence Update for Microsoft Defender Antivirus - KB2267602 (Version 1.397.556.0)` installed.

**Comment by TyIsI** [comment]
I didn't download with Edge, but I did scan my old Wintendo version and then installed the new version, both without issue.

Thanks all for all the effort!
